src/CodeGenerator/src/Generator/ResponseParser/JsonParser.php

```diff
@@ -230,11 +230,30 @@ private function parseResponseList(ListShape $shape, string $input, bool $requir
 
     private function parseResponseMap(MapShape $shape, string $input, bool $required): string
     {
+        $shapeValue = $shape->getValue();
+
         if (null === $locationName = $shape->getKey()->getLocationName()) {
-            throw new \RuntimeException('This is not implemented yet');
+            // This is a map
+
+            $body = '(function(array $json): array {
+                $items = [];
+                foreach ($json as $name => $value) {
+                    $items[$name] = CLASS::create($value);
```

Comment:
Hmm here, you presume that this is a Map of struct, which is not always the case. For instances, `Tags` are often a map of strings.

Looks at parseResponseList (which is very close to map, expect array's Key)

Reply:
Thank you